mirrord-layer/src/pod_api.rs

```diff
@@ -210,15 +228,22 @@ async fn create_ephemeral_container_agent(
 
     while let Some(status) = stream.try_next().await? {
         match status {
-            WatchEvent::Added(_) => break,
+            WatchEvent::Modified(pod) | WatchEvent::Added(pod) => {
+                if is_ephemeral_container_running(pod, &mirrord_agent_name) {
+                    debug!("container ready");
+                    break;
+                } else {
+                    debug!("container not ready yet");
+                }
+            }
             WatchEvent::Error(s) => {
                 error!("Error watching pod: {:?}", s);
                 break;
             }
             _ => {}
         }
     }
```

Comment:
Can we return LayerError here? Because if it's an error it just breaks out of the loop

Reply:
I thought about it and I think in worst case scenario it won't be able to connect to the agent then it will fail.